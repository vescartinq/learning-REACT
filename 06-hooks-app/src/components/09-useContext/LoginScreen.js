import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);

  let user = {
    id: 123,
    name: 'Victor',
  };

  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <button className="btn btn-primary" onClick={() => setUser(user)}>
        Login
      </button>
    </div>
  );
};
