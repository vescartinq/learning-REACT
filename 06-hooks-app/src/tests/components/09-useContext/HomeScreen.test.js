import React from 'react';
import { mount } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Tests in <HomeScreen />', () => {
  const user = {
    name: 'Victor',
    email: 'victor@gmail.com',
  };

  // SHALLOW solo renderiza componente principal, MOUNT renderiza todos los componentes, padres e hijos
  const wrapper = mount(
    <UserContext.Provider
      value={{
        user,
      }}
    >
      <HomeScreen />
    </UserContext.Provider>
  );

  test('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
