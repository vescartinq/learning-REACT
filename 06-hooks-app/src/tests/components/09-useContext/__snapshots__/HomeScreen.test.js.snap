// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <HomeScreen /> should render component 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Victor",
   "email": "victor@gmail.com"
}
  </pre>
</div>
`;
